From 09eea3958df50994153eb2933c607a6c5a820bca Mon Sep 17 00:00:00 2001
From: Mathieu Masson <mathieu.masson@plakar.io>
Date: Wed, 21 Oct 2026 10:37:25 +0200
Subject: [PATCH 5/5] pkg/ca-certificates: Update to Alpine 20261020.

* This is a fix for the eMudhra intermediate CA key compromise.

Signed-off-by: Mathieu Masson <mathieu.masson@plakar.io>
(cherry picked from commit 42bdb1db83933f746abade0eaa1e9266cb12810b)
---
 pkg/ca-certificates/Dockerfile | 6 +++++-
 pkg/ca-certificates/build.yml  | 1 +
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/pkg/ca-certificates/Dockerfile b/pkg/ca-certificates/Dockerfile
index ec09ba8..0297bea 100644
--- a/pkg/ca-certificates/Dockerfile
+++ b/pkg/ca-certificates/Dockerfile
@@ -1,6 +1,10 @@
 FROM linuxkit/alpine:6090baae063eb5023c9601966e88df831f789a70 as alpine
 
-RUN apk add ca-certificates
+# The private keys of eight eMudhra intermediates chaining to
+# emSign ECC Root CA - G3 are public. Alpine dropped the root in
+# ca-certificates 20261020-r0; fetch it from upstream until linuxkit/alpine
+# mirrors it, then go back to the plain mirror install.
+RUN apk add --no-cache --repositories-file /etc/apk/repositories.upstream 'ca-certificates>=20261020-r0'
 
 FROM scratch
 ENTRYPOINT []
diff --git a/pkg/ca-certificates/build.yml b/pkg/ca-certificates/build.yml
index bf158e3..617e188 100644
--- a/pkg/ca-certificates/build.yml
+++ b/pkg/ca-certificates/build.yml
@@ -1 +1,2 @@
 image: ca-certificates
+network: true
